src/dotnet-interactive/CommandLine/CommandLineParser.cs

```diff
@@ -529,7 +515,7 @@ static HttpPortRange ParsePortRangeOption(ArgumentResult result)
             SetUpFormatters(frontendEnvironment, startupOptions, TimeSpan.FromSeconds(15));
 
             kernel.DefaultKernelName = defaultKernelName;
-
+            services.AddKernel(kernel);
```

Comment:
Doing this here seems to blur some boundaries. I would expect the caller to this method to handle registration, and for this method to be non-side-affecting.

Reply:
that part of  code has always been like that as the kernel is created as side effect while executing the command line parser. the caller is passing in the ServiceCollection and merging it with the one used during asp.net configuration